llvm/lib/Transforms/Vectorize/VPlan.h

```diff
@@ -4247,7 +4247,8 @@ class VPlan {
   /// block with multiple predecessors (one for the exit via the latch and one
   /// via the other early exit).
   bool hasEarlyExit() const {
-    return ExitBlocks.size() > 1 || ExitBlocks[0]->getNumPredecessors() > 1;
+    return ExitBlocks.size() > 1 ||
+           (ExitBlocks.size() > 0 && ExitBlocks[0]->getNumPredecessors() > 1);
```

Comment:
```suggestion
           (ExitBlocks.size() == 1 && ExitBlocks[0]->getNumPredecessors() > 1);
```

In that case we must have a single exit block, right?